docs/design/byos.md

```diff
@@ -1,13 +1,142 @@
+Pools using this feature should be available to Azure DevOps Server customers.
+By making our hosted images available in the Azure gallery, customers get an equivalent to the Hosted pools without having to worry about multi-tenancy, a new billing relationship, etc.
+
+### VM specs and environment
+
+1. Customer wants more memory, more processor, or more IO than our native images.
+2. [Customer](https://github.com/MicrosoftDocs/vsts-docs/issues/2985) wants an NCv2 VM with particular instruction sets for machine learning. (It's niche enough that we won't stand up dedicated hosted pools, but broad enough to be very interesting for our business.)
+3. Customer wants additional storage attached to the VM. *(Real scenario from a medium-sized customer)*
+4. Customer wants to deploy to a private App Service. It's in a private VNET with no inbound connectivity. Today, this customer is forced to keep private agents standing by.
+
+### Preinstalled software
+
+1. Customer wants Windows 2016 with VS 2019 Preview. (We only offer certain combos like Win2016 + VS2017 and Win2019 + VS2019.)
+2. Customer wants to pin a specific set of tools and dependencies, preconfigured on the image.
+3. Customer wants extreme control over the exact OS build, settings, versions, and so on.
+
+### Agent reuse
+
+1. Customer wants to run several consecutive jobs on an agent to take advantage of things like incremental source and machine-level package caches. But, they don't want to run unnecessary VMs overnight when there's no load. They want to specify minimum and maximum # of agents associated with time ranges, but reuse agents within that span.
+2. Customer wants to run additional configuration or cache warmup before an agent beings accepting jobs. As additional agents are spun up, the customer has an opportunity to run a prep script that doesn't impact "pipeline runtime".
+
+### On-premises customers
+
+1. Customer wants to use Azure DevOps Server with elastic agent pools.
+
+## Industry review
+
+Similar problem spaces:
+- [Jenkins can use Azure agents](https://docs.microsoft.com/en-us/azure/jenkins/jenkins-azure-vm-agents) this way
+- [AppVeyor](https://www.appveyor.com/docs/enterprise/running-builds-on-azure/) offers instructions for solving a similar problem on several cloud providers
+- [GitLab CI/CD](https://gitlab.com/gitlab-org/gitlab-runner/blob/master/docs/configuration/autoscale.md) offers auto-scaling of builder containers using Docker Machine.
+
+Not offered:
+- [Travis CI](https://docs.travis-ci.com/user/enterprise/setting-up-travis-ci-enterprise/) offers an enterprise product that you can install on your own infrastructure. While you can choose what kind of VM to run it on, there's no elasticity.
+- [CircleCI](https://circleci.com/docs/2.0/aws/#nomad-clients) offers an on-your-infrastructure product. You must scale up and down the workers manually, there's no elasticity.
+
+## Solution
+
```

Comment:
Need to mention this is about Azure VMs (probably just VMSS?), but also consider AKS, any Kube, etc. AWS/GCB or is that just pool providers?